infrasim/init.py

```diff
@@ -94,16 +94,23 @@ def check_existing_workspace():
     if os.path.exists(config.infrasim_home):
         nodes = os.listdir(config.infrasim_home)
         if len(nodes) > 1:
-            print "There is node workspace existing.\n" 
-            print "If you want to remove it, please run:\n"
-            print "\"infrasim init -f \" "
-            exit()
+            return True
+        else:
+            return False
+
 
 def infrasim_init(node_type="dell_r730", skip_installation=True, force=False, target_home=None, config_file=None):
     try:
-        if force:
-            destroy_existing_nodes()
-            create_infrasim_directories()
+        if check_existing_workspace() is True:
+            if not force:
+                print "There is node workspace existing.\n"
+                print "If you want to remove it, please run:\n"
+                print "\"infrasim init -f \" "
+                exit()
```

Comment:
@chenge3 I'm thinking about this case, that infrasim_init() may be used as a library, typical character of a library function is:

- do its work;
- if can't raise exception

`exit()` is a dangerous action to cancel all stack. It can be a command line call action, but it's not proper to be here. Raise exception could be a better idea here, then this exception can be captured in cli and print corresponding message.